TLM/TLM/Manager/Impl/LaneArrowManager.cs

```diff
@@ -1,168 +1,171 @@
-using TrafficManager.State;
-using ColossalFramework;
-using TrafficManager.Geometry;
-using static TrafficManager.State.Flags;
-using TrafficManager.Traffic;
-using CSUtil.Commons;
-using TrafficManager.Geometry.Impl;
-
-namespace TrafficManager.Manager.Impl {
-	public class LaneArrowManager : AbstractGeometryObservingManager, ICustomDataManager<List<Configuration.LaneArrowData>>, ICustomDataManager<string>, ILaneArrowManager {
-		public const NetInfo.LaneType LANE_TYPES = NetInfo.LaneType.Vehicle | NetInfo.LaneType.TransportVehicle;
-		public const VehicleInfo.VehicleType VEHICLE_TYPES = VehicleInfo.VehicleType.Car;
-		public const ExtVehicleType EXT_VEHICLE_TYPES = ExtVehicleType.RoadVehicle &~ ExtVehicleType.Emergency;
-
-		public static readonly LaneArrowManager Instance = new LaneArrowManager();
-		
-		protected override void InternalPrintDebugInfo() {
-			base.InternalPrintDebugInfo();
-			Log._Debug($"- Not implemented -");
-			// TODO implement
-		}
-
-		public LaneArrows GetFinalLaneArrows(uint laneId) {
-			return Flags.getFinalLaneArrowFlags(laneId, true);
-		}
-
-		public bool SetLaneArrows(uint laneId, LaneArrows flags, bool overrideHighwayArrows = false) {
-			if (Flags.setLaneArrowFlags(laneId, flags, overrideHighwayArrows)) {
-				OnLaneChange(laneId);
-				return true;
-			}
-			return false;
-		}
-
-		public bool ToggleLaneArrows(uint laneId, bool startNode, LaneArrows flags, out LaneArrowChangeResult res) {
-			if (Flags.toggleLaneArrowFlags(laneId, startNode, flags, out res)) {
-				OnLaneChange(laneId);
-				return true;
-			}
-			return false;
-		}
-
-		protected void OnLaneChange(uint laneId) {
-			Services.NetService.ProcessLane(laneId, delegate (uint lId, ref NetLane lane) {
-				RoutingManager.Instance.RequestRecalculation(lane.m_segment);
-
-				if (OptionsManager.Instance.MayPublishSegmentChanges()) {
-					Services.NetService.PublishSegmentChanges(lane.m_segment);
-				}
-				return true;
-			});
-		}
-
-		protected override void HandleInvalidSegment(SegmentGeometry geometry) {
-			Flags.resetSegmentArrowFlags(geometry.SegmentId);
-		}
-
-		protected override void HandleValidSegment(SegmentGeometry geometry) {
-			
-		}
-
-		public void ApplyFlags() {
-			for (uint laneId = 0; laneId < NetManager.MAX_LANE_COUNT; ++laneId) {
-				Flags.applyLaneArrowFlags(laneId);
-			}
-		}
-
-		public override void OnBeforeSaveData() {
-			base.OnBeforeSaveData();
-			ApplyFlags();
-		}
-
-		public override void OnAfterLoadData() {
-			base.OnAfterLoadData();
-			Flags.clearHighwayLaneArrows();
-			ApplyFlags();
-		}
-
-		[Obsolete]
-		public bool LoadData(string data) {
-			bool success = true;
-			Log.Info($"Loading lane arrow data (old method)");
+﻿namespace TrafficManager.Manager.Impl {
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using ColossalFramework;
+    using CSUtil.Commons;
+    using Geometry.Impl;
+    using State;
+    using Traffic;
+    using static State.Flags;
```

Comment:
Might just be me, but I actually find the fully qualified using statements (with full namespace) easier to read as it makes it more obvious which stuff is from `TrafficManager`...

Reply:
This is StyleCop SA1200, either accept it or disable it, there is no middle ground. Doesn't bother me at all as this section is minimized 99% of the time.